app/controllers/likes_controller.rb

```diff
@@ -0,0 +1,25 @@
+class LikesController < ApplicationController
+  def new
+    @user = current_user
+  end
+
+  def create
+    @post = Post.find(params[:post_id])
+    @user = current_user
+    @like = Like.new
+    @like.user = @user
+    @like.post = @post
+    if @like.save
+      redirect_to user_post_url(@user, @post), notice: 'Post liked successfully.'
+    else
+      flash.now[:error] = 'Failed to like the post.'
+      render 'posts/show'
+    end
+  end
+
+  private
+
+  def comment_params
+    params.require(:comment).permit(:text)
+  end
```

Comment:
- [ **OPTIONAL** ] I think it would be better if you remove some unused code lines from your **likes_controller** for best practices and clean code reasons. I noticed that you added some code lines related to the **comments_controller** in the `likes_controller` such asline `20` to `24`,... You can only leave here these:
```ruby 
...
 def create
    @post = Post.find(params[:post_id])
    @user = current_user
    @like = Like.new
    @like.user = @user
    @like.post = @post
    if @like.save
      redirect_to user_post_url(@user, @post), notice: 'Post liked successfully.'
    else
      flash.now[:error] = 'Failed to like the post.'
      render 'posts/show'
    end
  end
  ...
```